### PR TITLE
NMS-16184: Upgrade to netty4 

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-16184

### CODE REVIEW

File: opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/AsyncBasicDetectorNettyImpl.java [hunk 1]
```diff
@@ -101,7 +83,7 @@ public AsyncBasicDetectorNettyImpl(final String serviceName, final int port, fin
     @Override
     public void dispose(){
         LOG.debug("calling dispose on detector {}", getServiceName());
-        m_factory.releaseExternalResources();
+        //m_factory.releaseExternalResources();
```
Comment (synqotik) [inline opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/AsyncBasicDetectorNettyImpl.java:86]: Delete commented-out line?

File: opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/DetectorHandlerNettyImpl.java [hunk 2]
```diff
@@ -39,115 +37,72 @@
...
-    public DetectFuture getFuture() {
-        return m_future;
-    }
-    */
-
     /** {@inheritDoc} */
     @Override
-    public void channelOpen(ChannelHandlerContext ctx, ChannelStateEvent event) throws Exception {
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
         LOG.debug("channelOpen()");
```
Comment (synqotik) [inline opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/DetectorHandlerNettyImpl.java:49]: Change log message to `channelActive()` ?

File: opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/DetectorHandlerNettyImpl.java [hunk 3]
```diff
@@ -39,115 +37,72 @@
...
-            ctx.getChannel().write(request);
+            ctx.channel().write(request);
         }
     }
 
     /** {@inheritDoc} */
     @Override
-    public void channelClosed(ChannelHandlerContext ctx, ChannelStateEvent event) throws Exception {
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         LOG.debug("channelClosed()");
```
Comment (synqotik) [inline opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/DetectorHandlerNettyImpl.java:59]: Change log message to `channelInactive()` ?

**Review by synqotik — APPROVED** [review]
LGTM. Approved, with a few small comments to consider

File: opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/AsyncBasicDetectorNettyImpl.java (hunk 1 above)
Comment (christianpape) [reply opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/AsyncBasicDetectorNettyImpl.java]: done.

File: opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/DetectorHandlerNettyImpl.java (hunk 2 above)
Comment (christianpape) [reply opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/DetectorHandlerNettyImpl.java]: done.

File: opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/DetectorHandlerNettyImpl.java (hunk 3 above)
Comment (christianpape) [reply opennms-provision/opennms-provision-api/src/main/java/org/opennms/netmgt/provision/support/DetectorHandlerNettyImpl.java]: done.

**Pull request merged**